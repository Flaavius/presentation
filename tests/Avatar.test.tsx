/* tslint:disable */
import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

import { Avatar } from "../src/Components/Avatar";

beforeEach(() => {
  //@ts-ignore
  global.fetch = jest.fn().mockImplementation(() => {
    let p = new Promise((res,rej) => {
      res({
        status: 200,
        blob: () => new Promise((res,rej) => {
          res('BLOBDATA');
        })
      });
    });
    return p;
  });
  //@ts-ignore
  global.URL.createObjectURL = jest.fn().mockImplementation((a) => { 
    return a;
  });
});

describe("Avatar", () => {
  it("Renderds like a boss", async (done) => {
    let avatar;
    act(() => {
      avatar = mount(<Avatar url="https://url-of-your-server.com/example/json" size={{ width: 50, height: 50 }} />);
    });

    //@ts-ignore
   expect(global.fetch).toHaveBeenCalledTimes(1);   
    //@ts-ignore
   expect(avatar).toMatchSnapshot();
    //@ts-ignore
    global.fetch.mockClear();
    done();
  }); 
});
