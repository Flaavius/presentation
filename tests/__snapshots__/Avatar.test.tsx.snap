// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Renderds like a boss 1`] = `
<Component
  size={
    Object {
      "height": 50,
      "width": 50,
    }
  }
  url="https://url-of-your-server.com/example/json"
>
  <div
    className="avatar  shadow"
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  >
    <div
      className="loader"
    />
  </div>
</Component>
`;
